narwhal/consensus/src/bullshark.rs

```diff
@@ -35,6 +38,33 @@ impl ConsensusProtocol for Bullshark {
         let round = certificate.round();
         let mut consensus_index = consensus_index;
 
+        // We must have stored already the parents of this certiciate!
+        if round > 0 {
+            let parents = certificate.header.parents.clone();
+            let store_parents: BTreeSet<&CertificateDigest> = state
+                .dag
+                .get(&(round - 1))
+                .expect("We should have the previous round!")
```

Comment:
What if we receive an old certificate and we don't have its parents? I mean currently this is possible in our protocol (it will get ignored in the sequencing process anyways). Now it seems that we'll crash which might not be desired

Reply:
Good catch -- I now print an error in case the round does not exist, but do not panic through `expect`.